cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/DefaultMacroEvaluator.java

```diff
@@ -32,12 +34,14 @@
   private final Map<String, String> resolvedArguments;
   private final long logicalStartTime;
   private final LogicalStartTimeMacro logicalStartTimeMacro;
+  private final SecureStore secureStore;
+  private final String namespace;
 
   private static final String LOGICAL_START_TIME_FUNCTION_NAME = "logicalStartTime";
   private static final String SECURE_FUNCTION_NAME = "secure";
 
   public DefaultMacroEvaluator(@Nullable WorkflowToken workflowToken, Map<String, String> runtimeArguments,
-                               long logicalStartTime) {
+                               long logicalStartTime, SecureStore secureStore, String namespace) {
```

Comment:
this class needs a namespace only for secure store, would have been nicer if we had `NamespacedSecureStore` implementation. @Nishith ?